flink-table/flink-table-runtime/src/main/java/org/apache/flink/table/runtime/functions/SqlJsonUtils.java

```diff
@@ -265,7 +265,13 @@ private static Object jsonQuery(
                             for (int i = 0; i < list.size(); i++) {
                                 final Object el = list.get(i);
                                 if (el != null) {
-                                    arr[i] = StringData.fromString(el.toString());
+                                    final String stringifiedEl;
```

Comment:
if there are `null` elements in the list, we shouldn't we print `null` rather than the object.

Reply:
No, it should not. `null` is different from a string containing null, we map JSON objects to SQL types here. In SQL we have a proper `null` representation.

Reply:
Maybe I am missing something. I was thinking of arrays of nullable objects. In this case the array element itself is validly a null. How do you think that should that look?
This looks like a valid scenario that Flink supports see :
[https://issues.apache.org/jira/browse/FLINK-30282](https://issues.apache.org/jira/browse/FLINK-30282)
[https://issues.apache.org/jira/browse/CALCITE-2464](https://issues.apache.org/jira/browse/CALCITE-2464)
[https://issues.apache.org/jira/browse/CALCITE-4085](https://issues.apache.org/jira/browse/CALCITE-4085)

Reply:
> Maybe in this case we cannot get nulls in the result as we are always querying for a none null predicates.

Why not? `{"items": ["itemId", null, "count"]}` is a valid array which should have a `null` at second position not `"null"`

Reply:
> > Maybe in this case we cannot get nulls in the result as we are always querying for a none null predicates.
> 
> Why not? `{"items": ["itemId", null, "count"]}` is a valid array which should have a `null` at second position not `"null"`

Yes I agree with your example. So are we saying that an array of objects or an array of arrays that is nullable should have a json null for null elements? If so that would be what we are looking for. 

I think it would be worth adding some more unit tests 
- arrays of primitives, when 1 or more of the elements are null
- arrays of arrays when 1 or more of the elements are null 
- arrays of objects when 1 or more of the elements are null 
- A 3 deep array of arrays with nullable elements in the deepest child, to show it is coming out consistently at that level as well.   
WDYT?

Reply:
> arrays of arrays when 1 or more of the elements are null
> arrays of objects when 1 or more of the elements are null

Those are not different from 1)

> A 3 deep array of arrays with nullable elements in the deepest child, to show it is coming out consistently at that level as well.

We do not support multiple levels of types conversions. We support only `ARRAY<STRING>` at most. `RETURNING ARRAY<ARRAY<...>>` is unsupported.

Reply:
Added some `nulls` to the tests